lib/kubeclient/common.rb

```diff
@@ -63,6 +70,9 @@ def initialize_client(
       @ssl_options = ssl_options
       @auth_options = auth_options
       @socket_options = socket_options
+      # Allow passing partial timeouts hash, without unspecified
+      # @timeouts[:foo] == nil resulting in infinite timeout.
```

Comment:
With other options, you either get FOO_DEFAULTS or replace it entirely.
If you pass less keys than  FOO_DEFAULTS had, it's treated as nil, which is generally buggy, in this case would *silently* give infinite timeout which I felt too subtle, so broke the pattern.

I'll send separate PR to do same with other options (and drop this comment).